ext/gmp/gmp.c

```diff
@@ -515,18 +460,24 @@ static zend_result gmp_do_operation(uint8_t opcode, zval *result, zval *op1, zva
 
 static int gmp_compare(zval *op1, zval *op2) /* {{{ */
 {
-	zval result;
-
-	gmp_cmp(&result, op1, op2, /* is_operator */ true);
+	mpz_ptr gmp_op1, gmp_op2;
+	bool status = false;
 
-	/* An error/exception occurs if one of the operands is not a numeric string
-	 * or an object which is different from GMP */
-	if (EG(exception)) {
-		return 1;
+	status = gmp_zend_parse_arg_into_mpz_ex(op1, &gmp_op1, 1, true);
+	if (!status || EG(exception)) {
```

Comment:
Is the `|| EG(exception)` necessary? And if yes, it should be done too below; but I don't think it is.

Reply:
Ah no I didn't update this properly when I changed my implementation